File: 2.arrays/lesson_2_cyclic_rotation.py
def solution(A, K):
    A_size = len(A)
    B = A[:]
    if(A_size == K):
        return A
    else:
        for i, element in enumerate(A):
            new_index = (K + i) % A_size
            B[new_index] = element
        return B

File: 2.arrays/lesson_2_cyclic_rotation.js
function solution(A, K) {
    let a = A.length
    let rotate = K % a
    let rotatedA = new Array(a)

    for(let i = 0; i < a; i++) {
        let j = i + rotate
        if(j >= a) {
            j -= a
        }
        rotatedA[j] = A[i]
    }
    return rotatedA
}